### DISCUSSION TITLE
Any plans to support jetpack compose?

### DISCUSSION THREAD

**Discussion by sarimmehdi** [sceneview/sceneview#169] [Ideas]
Even a workaround would be appreciated. Not a big fan of XML, personally.

**Comment by ThomasGorisse** [comment]
Yes! SceneView has been Compose-first since v2.0, and the latest **v4.0.0-rc.1** makes it even better. Here's a minimal example:

```kotlin
implementation("io.github.sceneview:sceneview:4.0.0-rc.1")

@Composable
fun My3DViewer() {
    SceneView {
        val model = rememberModelInstance(modelLoader, "models/mymodel.glb")
        if (model != null) {
            ModelNode(modelInstance = model)
        }
    }
}
```

For AR, use `ARSceneView { }` from the `arsceneview` artifact. All nodes are declared as composables inside the content block. See `samples/android-demo` for 19 working demos.
